applications/flash_attention_v2/kernel/xe_flash_attn_gemm.hpp

```diff
@@ -346,30 +365,31 @@ class GemmUniversalAttention {
         Tensor tSr = make_tensor<ElementAccumulator>(Shape<Int<Vec>, Int<FragsM>, Int<FragsN>>{});
         clear(tSr);
         // 3) Perform GEMM S = Q*K
-        collective_mma.mmaQK(tSr, gQ,  gK(_, _, nblock_limit - 1, _), tSr, ceil_div(head_size_qk, QK_BLK_K), mainloop_params);
+        collective_mma.mmaQK(tSr, gQ,  gK(_, _, nblock_new - 1, _), tSr, ceil_div(head_size_qk, QK_BLK_K), mainloop_params, false);
         // we only need one block ahead, there is enough gap to prefetch it while doing softmax. because the gap between the two MMA is big,
         // prefetching it the same way as cutlass K matrix does not make sense
-        prefetch(tiled_prefetch_v, pVgV(_, _, _ , nblock_limit - 1));
+        prefetch(tiled_prefetch_v, pVgV(_, _, _ , nblock_new - 1));
         // mask the elements of each tile where j > i
         const int item_id = thread_idx % SubgroupSize;
-        int col_idx = item_id + (nblock_limit - 1) * QK_BLK_N;
+        int col_idx = item_id + (nblock_new - 1) * QK_BLK_N;
         CUTLASS_PRAGMA_UNROLL
         for (int n = 0; n < FragsN; n++, col_idx += get<1>(MmaAtomShape())) { // 4
           CUTLASS_PRAGMA_UNROLL
           for (int m = 0; m < FragsM; m++) { // 2
             int row_idx = m * Vec + seq_coord;
             CUTLASS_PRAGMA_UNROLL
             for (int row = 0; row < Vec; row++, row_idx++) { // 8
-              if (col_idx > row_idx)
-                tSr(row, m, n) = -INFINITY;
+                if (col_idx > row_idx) {
+                    tSr(row, m, n) = -INFINITY;
+                }
             }
           }
         }
 
         CollectiveSoftmaxEpilogue softmax(params.softmax);
-        softmax((nblock_limit - 1) == 0, tSr, max_reg, sum_reg, out_reg);
+        softmax(is_first_block, tSr, max_reg, sum_reg, out_reg);
```

Comment:
Explained above that why this should stay as it is. This condition in almost all cases will translate to false. This condition check if any n block has run before, and whether as a cached or non-cached part.

Reply:
The condition will always be false only if there is cached KV. Otherwise, checks if its the first block or not.